testing/src/main/java/io/stargate/it/cql/AdditionalPortsTest.java

```diff
@@ -0,0 +1,105 @@
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+
+@ExtendWith(CqlSessionExtension.class)
+@StargateSpec(nodes = 2, parametersCustomizer = "buildParameters")
+@CqlSessionSpec(contactPointResolver = EmptyContactPointResolver.class, createSession = false)
+public class AdditionalPortsTest extends BaseOsgiIntegrationTest {
+
+  public static final List<Integer> ADDITIONAL_PORTS = ImmutableList.of(29042, 39042);
+
+  // The tests are going to set the contact points explicitly
+  public static class EmptyContactPointResolver implements ContactPointResolver {
+
+    @Override
+    public List<InetSocketAddress> resolve(ExtensionContext context) {
+      return Collections.emptyList();
+    }
+  }
+
+  @ParameterizedTest
+  @DisplayName("Connects to stargate using the main and additional port values")
+  @MethodSource("allPorts")
+  public void connectToMainAndAdditionalPorts(
+      int port, CqlSessionBuilder builder, StargateEnvironmentInfo stargate) {
+    builder.addContactPoint(new InetSocketAddress(stargate.nodes().get(0).seedAddress(), port));
+    CqlSession session = builder.build();
+
+    Iterator<Node> nodes = session.getMetadata().getNodes().values().iterator();
+
+    Node localNode = nodes.next();
+    Row localRow =
+        session
+            .execute(
+                SimpleStatement.builder("SELECT * FROM system.local").setNode(localNode).build())
+            .one();
+    assertThat(localRow).isNotNull();
+    assertThat(localRow.getInetAddress("rpc_address"))
+        .isEqualTo(
+            localNode.getBroadcastRpcAddress().map(InetSocketAddress::getAddress).orElse(null));
+    if (backend.isDse()) {
+      assertThat(localRow.getInt("native_transport_port")).isEqualTo(port);
+    }
+
+    ResultSet rs =
+        session.execute(
+            SimpleStatement.builder("SELECT * FROM system.peers").setNode(localNode).build());
+    List<Row> rows = rs.all();
+
+    assertThat(rows).hasSizeGreaterThan(0);
+    List<InetAddress> rpcAddresses = new ArrayList<>();
+    rows.forEach(
+        row -> {
+          rpcAddresses.add(row.getInetAddress("rpc_address"));
+          if (backend.isDse()) {
+            assertThat(row.getInt("native_transport_port")).isEqualTo(port);
+          }
+        });
+    List<InetAddress> expectedRpcAddresses =
+        Streams.stream(nodes)
+            .map(n -> n.getBroadcastRpcAddress().map(InetSocketAddress::getAddress).orElse(null))
+            .collect(Collectors.toList());
+    assertThat(rpcAddresses).containsExactlyInAnyOrderElementsOf(expectedRpcAddresses);
+  }
+
+  public static List<Integer> allPorts() {
+    return ImmutableList.<Integer>builder().addAll(ADDITIONAL_PORTS).add(9043).build();
```

Comment:
Why is 9043 here and not in the `ADDITIONAL_PORTS` list?

Reply:
It's the "main" port.

Reply:
Pulling this from the environment configuration now: https://github.com/stargate/stargate/pull/1115/commits/fdb97b5c73020ff98e67c94f0d7c9d776a21686c